app/uk/gov/hmrc/customs/inventorylinking/export/controllers/InventoryLinkingExportController.scala

```diff
@@ -65,20 +65,29 @@ class InventoryLinkingExportController @Inject()(override val authConnector: Inv
 
   def post(): Action[AnyContent] = validateHeaders().async(bodyParser = xmlOrEmptyBody) {
     implicit request =>
+      val maybeBadgeIdentifier: Option[BadgeIdentifier] = extractBadgeIdentifier(request)
```

Comment:
Badge identifier is mandatory - so I think it should not be an option here and in subsequent uses/methods -. I think it gets quite confusing when you start to pass things around that are mandatory as an option.